src/moi.jl

```diff
@@ -289,6 +289,88 @@ end
 accept_vector_set(::ProductMode{T}, ::Complement) where {T} = nothing
 accept_vector_set(::MixedMode{T}, ::Complement) where {T} = nothing
 
+function get_variable_complement(primal_model, dual_model, primal_con, dual_con)
+    return error(
+        "An internal error with variable complements occurred. Likely, your problem type does not yet support consideration of constrained variables.",
+    )
+end
+
+function get_variable_complement(
+    primal_model,
+    dual_model,
+    primal_con::MOI.ConstraintIndex{Fp,Sp},
+    dual_con::MOI.ConstraintIndex{Fd,Sd},
+) where {
+    Fp<:MOI.VariableIndex,
+    Sp<:Union{MOI.LessThan{T},MOI.GreaterThan{T}},
+    Fd,
+    Sd<:Union{MOI.LessThan{T},MOI.GreaterThan{T}},
+} where {T}
+    primal_variable =
+        MOI.get(primal_model, MOI.ConstraintFunction(), primal_con)
+    primal_set = MOI.get(primal_model, MOI.ConstraintSet(), primal_con)
+
+    @assert MOI.constant(primal_set) == 0 "Unexpected variable bound"
+
+    dual_func = MOI.get(dual_model, MOI.ConstraintFunction(), dual_con)
+    dual_set = MOI.get(dual_model, MOI.ConstraintSet(), dual_con)
```

Comment:
you need MOI.copy just like in get_canonical_complement

Reply:
Now done, but only for the dual_func. This is the only necessary one, right? 
I'm asking because the copy is done for sets etc. in other functions (but that should be handled by set_with_zero, right? 

PS: This was a big one, so I'm a bit surprised it didn't affect testing...